enumerable.rb

```diff
@@ -0,0 +1,130 @@
+module Enumerable
+  def my_each
+    return to_enum(my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    rreturn to_enum(my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(my_select) unless block_given?
+
+    result = []
+    my_each do |item|
+      result << item if yield(item)
+    end
+    result
+  end
+
+  def my_all?(param = nil)
+    result = true
```

Comment:
* [ ] should have working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_all? 

- [ ] when no block or argument is given returns true only when none of the collection members are false or nil, if the mentioned condition is not met it, should return false

```ruby     
[nil, false, true, []].my_all? == [nil, false, true, []].all? => false
```
- Yours is giving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/79289276-10281080-7e86-11ea-8a2a-d9482b50498f.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688266-67148700-6f65-11ea-82e5-cbef16444817.png)

Reply:
done